src/ws2st/ws2st.cpp

```diff
@@ -0,0 +1,115 @@
+#include "ws2st.hpp"
+
+// Apply a custom category to all command-line options so that they are the
+// only ones displayed.
+static llvm::cl::OptionCategory WashS2STCategory("Wash S2S Translator");
+
+// CommonOptionsParser declares HelpMessage with a description of the common
+// command-line options related to the compilation database and input files.
+// It's nice to have this help message in all tools.
+static cl::extrahelp CommonHelp(CommonOptionsParser::HelpMessage);
+
+// A help message for this specific tool can be added afterwards.
+static cl::extrahelp MoreHelp("\nThis is the WaSH source-to-source translator tool.\n");
+
+int main(int argc, const char **argv) {
+
+    // argv == wash/old --> copy over the src and original wash backend no changes made
+    //      == wash/isb --> copy over the src and new wisb backend and do the enum changes ets
+    //      == openmp   --> add in linking with OpenMP
+
+    if (argc < 2) {
+        std::cerr << "Usagae: " << argv[0] << " /path/to/source -- [other compiler options]" << std::endl;
+        return 1;
+    }
+
+    std::vector<std::filesystem::path> files = wash::files::copy_wash_source(argv[1]);
+    std::vector<char *> new_args;
+
+    char* cstr = new char[std::strlen(argv[0]) + 1];
```

Comment:
Can we use `std::string`s instead of `char` arrays? Mentioned elsewhere but using the `new` keyword can easily lead to leaks.

Reply:
Yeah this is weird. The clang tool constructor explicitly needed a `char **` so this was at the time easier to construct than trying to convert a vector of std strings to a char**

Reply:
fwiw I do delete these at the end of the function so they don't leak.

Reply:
Yep no leaks here in its current state. `unique_ptr`s just would allow us to never even worry about deleting them

Reply:
Moved to be a vector of strings and transformed into char**